packages/Uploader/src/helpers.js

```diff
@@ -0,0 +1,122 @@
+import uuidv4 from "uuid/v4";
+import superagent from "superagent";
+import types from "./types";
+
+export function fileSizeUnitsToHumanReadableFormat(size) {
+  const aMultiples = ["KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB"];
+  let nMultiple = 0;
+  let sOutput = "";
+  // this calculate Kibebytes, Mebibyes, etc.
+  // But are being label as KB and MB for easier understanding.
+  for (let nApprox = size / 1024; nApprox > 1; nApprox /= 1024, nMultiple++) {
+    // fixed to 2 decimals but we could add more if need it
+    sOutput = `${nApprox.toFixed(2)} ${aMultiples[nMultiple]}`;
```

Comment:
Seems odd to keep recalculating this.  Maybe a `while` loop could be employed to get the `nApprox` and `nMultiple` you need, then just return this.

Reply:
no, understanding exactly what you mean, this was a port from our previous code, and we need to calculate for each field uploaded, and recalculated it if the files changes.

Reply:
Okay, 🙈